pkg/operator/controller/controller_test.go

```diff
@@ -1314,6 +1264,29 @@ func setDeploymentsReady(args *args) bool {
 	return running
 }
 
+func setDeploymentsDegraded(args *args) {
+	resources, err := getAllResources(args.reconciler)
+	Expect(err).ToNot(HaveOccurred())
+
+	for _, r := range resources {
+		d, ok := r.(*appsv1.Deployment)
+		if !ok {
+			continue
+		}
+
+		d, err := getDeployment(args.client, d)
+		Expect(err).ToNot(HaveOccurred())
+		if d.Spec.Replicas != nil {
+			d.Status.Replicas = int32(0)
+			d.Status.ReadyReplicas = d.Status.Replicas
+			err = args.client.Update(context.TODO(), d)
+			Expect(err).ToNot(HaveOccurred())
+		}
+
+	}
+	doReconcile(args)
```

Comment:
should conditions be verified here?

Reply:
I am assuming you are asking if I should verify if the degraded condition is true. The problem is depending on the state of the CR, degraded can be either true or false (if I am deploying degraded is always false, but if I am deployed already, then degraded will be true). I guess I can check if I am deploying or not and check based on that.